### DISCUSSION TITLE
Error when backfilling

### DISCUSSION THREAD

**Discussion by luancaarvalho** [apache/airflow#25071] [General]
### Apache Airflow version

2.3.0

### What happened

Hi !
I'm trying to backfill my dag that searches for an attachment in a mailbox and send it to s3 but on any date even after my dag has done all the tasks completely they keep running infinite.

Ps: I set max_active_runs to 5 looking to avoid errors in mailbox requests

### What you think should happen instead

I think that after a one dag run completes another has to start.

### How to reproduce

I'm running this command:

airflow dags backfill --reset-dagruns --start-date 2020-03-30 --end-date 2021-01-01 my_dag_id

### Operating System

windows

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

I'm using the latest deploy of helm withou many changes just configuring volumes and setting env vars

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is not reproducibleairflow bug. There is pretty much 0 information here that could help anyone to help you to solve your problem. Converted it into discussion. If you need help, you need to make a deeper investigation, provide some logs, and other evidences of what's happening. People here help in their free time usually so you need ot make sure you provide enough information and did your part on analysing your problem before you create an issue (and you need to have reproducible case that anyone can reproduce).

Plesase do more exploration, gather observations - look at the logs, try to find any problems, debug, and come up with some more diagnosable issues if you need help in solving your issue.
